config/core/deployments/webhook-hpa.yaml

```
@@ -0,0 +1,51 @@
+# Copyright 2020 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: webhook
+  namespace: knative-serving
+  labels:
+    serving.knative.dev/release: devel
+spec:
+  minReplicas: 1
+  maxReplicas: 5
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: webhook
+  metrics:
+    - type: Resource
+      resource:
+        name: cpu
+        # Percentage of the requested CPU
+        targetAverageUtilization: 100
+
+
+---
+
+# Webhook PDB.
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: webhook-pdb
+  namespace: knative-serving
+  labels:
+    serving.knative.dev/release: devel
+spec:
+  maxUnavailable: 20%
```

I haven't played with PodDisruptionBudgets so some questions

1) Why 20% maxUnavailable
2) What does this mean if there's only one replica

---

Hi @dprotaso , I was not sure what should be the best values for these, so I just mimicked the setup for activator. I do realize they are different, so if you have a better idea for the values here, please just let me know, and I would be happy to make the adjustments. Thanks!

---

So @vagababov and I chatted about this - `maxUnavailable` when we have one replica will allow disruption to happen. So we'll need to use `minAvailable: 80%`

See for more details: https://github.com/knative/serving/pull/9548

```suggestion
  minAvailable: 80%
```

---

To clarify the reason why it allows disruption is because of how K8s does calculations and rounds.

ie. maxUnavailable with 20% of 1 ends up equaling `maxUnavailable = 1` when subtracted against our replica count will be `0`

---

Really appreciate your detailed clarification @dprotaso ! It was very helpful.